envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -116,10 +116,29 @@ fun Value?.toIncoming(): Incoming {
         permissionsEnabled = endpointsField != null,
         healthCheck = this?.field("healthCheck").toHealthCheck(),
         roles = this?.field("roles")?.list().orEmpty().map { Role(it) },
-        timeoutPolicy = this?.field("timeoutPolicy").toIncomingTimeoutPolicy()
+        timeoutPolicy = this?.field("timeoutPolicy").toIncomingTimeoutPolicy(),
+        unlistedEndpointsPolicy = this?.field("unlistedEndpointsPolicy").toUnlistedEndpointsPolicy()
     )
 }
 
+fun Value?.toUnlistedEndpointsPolicy() = this?.stringValue
+    ?.takeIf { it.isNotEmpty() }
+    ?.let { when (it) {
+        "log" -> Incoming.UnlistedEndpointsPolicy.LOG
+        "blockAndLog" -> Incoming.UnlistedEndpointsPolicy.BLOCKANDLOG
+        else -> throw NodeMetadataValidationException("Invalid unlistedEndpointsPolicy value: $it")
+    } }
+    ?: Incoming.UnlistedEndpointsPolicy.BLOCKANDLOG
+
+fun Value?.toUnlistedClientsPolicy() = this?.stringValue
```

Comment:
Code looks the same for `toUnlistedEndpointsPolicy` and `toUnlistedClientsPolicy`. Do we want to keep both policies in separate classes in case of future changes?

Reply:
For now, I don't see a use case for separate policy sets on different levels. I'd consolidate these two in the sake of simplicity. We can extend and eventually separate in the future.

Reply:
yes, the reason was that we can differentiate these two policy sets in the future. I can merge it to one for now.

Reply:
done